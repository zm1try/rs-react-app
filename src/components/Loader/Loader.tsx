export const Loader = () => {
  return <p className={'loading'}>Loading...</p>;
};
