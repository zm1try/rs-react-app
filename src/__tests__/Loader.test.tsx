import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Loader } from '@/components/Loader/Loader.tsx';

describe('Loader', () => {
  it('renders the loading text', () => {
    const { getByText } = render(<Loader />);
    expect(getByText('Loading...')).toBeInTheDocument();
  });

  it('has the correct class name', () => {
    const { container } = render(<Loader />);
    const paragraph = container.querySelector('p');
    expect(paragraph).toHaveClass('loading');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
