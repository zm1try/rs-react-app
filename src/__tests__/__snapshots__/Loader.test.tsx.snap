// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader > matches the snapshot 1`] = `
<DocumentFragment>
  <p
    class="loading"
  >
    Loading...
  </p>
</DocumentFragment>
`;
